### PR TITLE
Date decode from integer values

### PR DESCRIPTION
**Pull request by Frizlab**
Adds support for decoding `Swift.Date` from SQLite integer values (#16).

### CODE REVIEW

File: Sources/SQLiteNIO/SQLiteDataConvertible.swift [hunk 1]
```diff
@@ -122,7 +122,15 @@ extension Bool: SQLiteDataConvertible {
 
 extension Date: SQLiteDataConvertible {
     public init?(sqliteData: SQLiteData) {
-        guard case .float(let value) = sqliteData else {
+        let value: Double
+        /* We have to retrieve floats and integers, because apparently SQLite
```
Comment (tanner0101) [inline Sources/SQLiteNIO/SQLiteDataConvertible.swift:126]: Using `//` for inline comments, please.

File: Tests/SQLiteNIOTests/NIOSQLiteTests.swift [hunk 2]
```diff
@@ -28,6 +28,20 @@ final class SQLiteNIOTests: XCTestCase {
         XCTAssertEqual(Date(sqliteData: rows[0].column("date")!)?.description, date.description)
     }
 
+    func testTimestampStorageInDateColumnIntegralValue() throws {
+        let conn = try SQLiteConnection.open(storage: .memory, threadPool: self.threadPool, on: self.eventLoop).wait()
+        defer { try! conn.close().wait() }
+
+        let date = Date(timeIntervalSince1970: 42)
+        /* This is how a column of type .date is crated when using Vapor’s
```
Comment (tanner0101) [inline Tests/SQLiteNIOTests/NIOSQLiteTests.swift:36]: Same here.

**Comment by Frizlab** [comment]
Will do that tomorrow. Should I amend and force push or just add new commits is ok?

**Comment by tanner0101** [comment]
Original comment:
---
Technically SQLite does not have support for other types than a very few, which do not include `DATE`.
However, it is possible to create a column typed as `DATE`, in which case anything can be put in it.
Users would expect to retrieve a valid date when putting an integer as a value of such a column, especially since adding a float integral value returns an int value.

This problems also begs the question: what about the `Double` and `Float` cases? Shouldn't we also handle integers returned by SQLite as valid doubles or floats, using the same implementation as I did in this PR? Which BTW would reduce the implementation for Date to a simple:
```swift
guard let value = Double(sqliteData: sqliteData) else {
    return nil
}
self.init(timeIntervalSince1970: value)
```

**Comment by tanner0101** [comment]
These changes are now available in [1.0.0-rc.1.1](https://github.com/vapor/sqlite-nio/releases/tag/1.0.0-rc.1.1)

**Pull request merged**